aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -628,10 +627,46 @@ def _call_route(self, route: Route, args: Dict[str, str]) -> ResponseBuilder:
                         status_code=500,
                         content_type=content_types.TEXT_PLAIN,
                         body="".join(traceback.format_exc()),
-                    )
+                    ),
+                    route,
                 )
+
             raise
 
+    def not_found(self, func: Callable):
+        return self.exception_handler(NotFoundError)(func)
+
+    def exception_handler(self, exc_class: Type[Exception]):
+        def register_exception_handler(func: Callable):
+            self._exception_handlers[exc_class] = func
+
+        return register_exception_handler
+
+    def _lookup_exception_handler(self, exp_type: Type) -> Optional[Callable]:
+        # Use "Method Resolution Order" to allow for matching against a base class
```

Comment:
Last ask, could you have a test for this? issubclass might work here, since `mro` could be extensive as we don't know how many inheritances one created.

Reply:
@heitorlessa there is a test `test_exception_handler_service_error` which registers a `ServiceError` handler and raises a `InternalServerError`